sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTest.java

```diff
@@ -1119,42 +1121,45 @@ public void testStreamingWriteWithWindowFn() throws Exception {
     FakeBigQueryServices fakeBqServices = new FakeBigQueryServices()
         .withDatasetService(datasetService);
 
-    List<TableRow> inserts = new ArrayList<>();
+    List<Integer> inserts = new ArrayList<>();
     for (int i = 0; i < 10; i++) {
-      inserts.add(new TableRow().set("name", "number" + i).set("number", i));
+      inserts.add(i);
     }
 
     // Create a windowing strategy that puts the input into five different windows depending on
     // record value.
-    WindowFn<TableRow, PartitionedGlobalWindow> window = new PartitionedGlobalWindows(
-        new SerializableFunction<TableRow, String>() {
+    WindowFn<Integer, PartitionedGlobalWindow> window = new PartitionedGlobalWindows(
+        new SerializableFunction<Integer, String>() {
           @Override
-          public String apply(TableRow value) {
-            try {
-              int intValue = (Integer) value.get("number") % 5;
-              return Integer.toString(intValue);
-            } catch (NumberFormatException e) {
-              fail(e.toString());
-            }
-            return value.toString();
+          public String apply(Integer i) {
+            return Integer.toString(i % 5);
           }
         }
     );
 
-    SerializableFunction<BoundedWindow, String> tableFunction =
-        new SerializableFunction<BoundedWindow, String>() {
+    SerializableFunction<ValueInSingleWindow<Integer>, String> tableFunction =
```

Comment:
Actually this business with ValueInSingleWindow looks a little awkward. I can't suggest something much better (especially given that we want to later extend it to support side inputs and such), so I am personally OK with it, but please get an additional person to sign off on it, e.g. Davor.